custom/icds_reports/static/js/angular-services/baseControllers.service.js

```diff
@@ -233,6 +233,26 @@ window.angular.module('icdsApp').factory('baseControllersService', function() {
                 }
             };
 
+            //Creating a promise, which resolves only after map is rendered (wait until canvas is rendered in dom)
+            // Reference: https://stackoverflow.com/a/47776379/12839195 (using timeout instead of rafAsync)
+            vm.checkElement = function (selector) {
```

Comment:
nit: since this function became coupled with the maps logic it could probably use a different name. e.g. `waitForMapIfNecessary`. And you could also not bother passing in `selector` and just hard-code it at the top of the function.

Reply:
ok